python/pyarrow/parquet.py

```diff
@@ -1493,15 +1493,16 @@ def __init__(self, path_or_paths, filesystem=None, filters=None,
                 single_file = path_or_paths[0]
         else:
             if _is_path_like(path_or_paths):
-                path = str(path_or_paths)
+                path_or_paths = str(path_or_paths)
                 if filesystem is None:
                     # path might be a URI describing the FileSystem as well
                     try:
-                        filesystem, path = FileSystem.from_uri(path)
+                        filesystem, path_or_paths = FileSystem.from_uri(
+                            path_or_paths)
```

Comment:
Ah, good catch. So we were passing below still the original `path_or_paths` URI to the dataset constructor (instead of the non-URI path returned by from_uri), but also passing the filesystem inferred from the URI here.

Reply:
Yup.